python/ray/experimental/serve/kv_store_service.py

```diff
@@ -0,0 +1,174 @@
+import json
+from abc import ABC
+
+import ray
+import ray.experimental.internal_kv as ray_kv
+from ray.experimental.serve.utils import logger
+
+
+class NamespacedKVStore(ABC):
+    """Abstract base class for a namespaced key-value store.
+
+    The idea is that multiple key-value stores can be created while sharing
```

Comment:
Don't database client libraries do this anyways?

Reply:
One plan is to fate-share with `ray` by using the `ray` GCS as the state storage system, which requires us to use redis. So we might as well roll our own namespacing solution.

Reply:
Sure that's fine, but the sharing a database connection part is not useful because it should be handled by the library. We just need a thin wrapper that adds the namespace logic.

Reply:
I should clarify in the doc. "sharing a database connection" -> "sharing the same storage system"